training/src/main/java/wolox/training/models/User.java

```diff
@@ -35,15 +37,17 @@
     @Column(nullable = false)
     private LocalDate birthdate;
 
-    @ManyToMany(cascade = CascadeType.ALL)
+    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH})
```

Comment:
You are really close! The only problem here is PERSIST. Because for the cases where the Book does already exist, it'll try persisting again

Reply:
fixed